pipeline/data_merging/brca_pseudonym_generator.py

```diff
@@ -165,89 +168,153 @@ def get_synonyms(x, hgvs_proc, syn_ac_dict):
 @click.option('--resources', help="path to directory containing reference sequences")
-def main(input, output, pkl, log_path, config_file, resources):
+@click.option('--processes', type=int, help='Number of processes to use for parallelization', default=8)
+def main(input, output, log_path, config_file, resources, processes):
     utils.setup_logfile(log_path)
 
     cfg_df = config.load_config(config_file)
 
-    syn_ac_dict = { x[config.SYMBOL_COL] : x[config.SYNONYM_AC_COL].split(';') for _, x in cfg_df.iterrows()}
-    cdna_default_ac_dict = { x[config.SYMBOL_COL] : x[config.HGVS_CDNA_DEFAULT_AC] for _, x in cfg_df.iterrows()}
+    syn_ac_dict = {r[config.SYMBOL_COL]: r[config.SYNONYM_AC_COL].split(';') for _, r in cfg_df.iterrows()}
+    cdna_default_ac_dict = {r[config.SYMBOL_COL]: r[config.HGVS_CDNA_DEFAULT_AC] for _, r in cfg_df.iterrows()}
+    strand_dict = { r[config.SYMBOL_COL] : r[config.STRAND_COL] for _, r in cfg_df.iterrows() }
 
     hgvs_proc = HgvsWrapper()
 
+    logging.info("Loading data from {}".format(input))
     df = pd.read_csv(input, sep='\t')
 
-    df[VAR_OBJ_FIELD] = df.apply(lambda x: VCFVariant(x[CHR_COL], x[POS_COL], x[REF_COL], x[ALT_COL]), axis=1)
+    df[VAR_OBJ_FIELD] = df.apply(lambda r: VCFVariant(r[CHR_COL], r[POS_COL], r[REF_COL], r[ALT_COL]), axis=1)
+
+    logging.info("Converting variants to hgvs objects")
+
+    df[TMP_HGVS_HG38] = df[VAR_OBJ_FIELD].apply(
+        lambda v: v.to_hgvs_obj(hgvs_proc.contig_maps[HgvsWrapper.GRCh38_Assem]))
+
+    logging.info("Normalize genomic representation")
+
+    df = utils.parallelize_dataframe(df, _normalize_genomic_fnc(TMP_HGVS_HG38, TMP_HGVS_HG38, True, strand_dict), processes)
+    df = utils.parallelize_dataframe(df, _normalize_genomic_fnc(TMP_HGVS_HG38, TMP_HGVS_HG38_LEFT_ALIGNED, False, strand_dict),
+                                     processes)
+
+    df[GENOMIC_HGVS_HG38_COL] = df[TMP_HGVS_HG38].apply(str)
+
+    logging.info("Compute hg37 representation of internal representation")
+    var_objs_hg37 = convert_to_hg37(df[VAR_OBJ_FIELD], resources)
+
+    logging.info("Compute hg37 normalized representation of internal")
+    # normalizing again for the hg37 representation. An alternative would be to convert the normalized hg38 representation to hg37.
+    # If we use crossmap, we would need a way to convert the VCF like representation back to an hgvs object, which we currently
+    # are unable to do properly. That is, we can use VCFVariant.to_hgvs_obj, however, structural variants will be converted
+    # to delins, losing information if a variant was e.g. a del, ins, or dup.
+
+    df[TMP_HGVS_HG37] = pd.Series(
+        [v.to_hgvs_obj(hgvs_proc.contig_maps[HgvsWrapper.GRCh37_Assem]) for v in var_objs_hg37])
+
+    df = utils.parallelize_dataframe(df, _normalize_genomic_fnc(TMP_HGVS_HG37, GENOMIC_HGVS_HG37_COL, True, strand_dict), processes)
+    df[GENOMIC_HGVS_HG37_COL] = df[GENOMIC_HGVS_HG37_COL].apply(str)
```

Comment:
same as above -- how come it takes two immediate runs of _normalize_genomic_fnc to get desired output?